autoload/test/java/maventest.vim

```diff
@@ -24,8 +24,20 @@ function! test#java#maventest#build_position(type, position) abort
 endfunction
 
 function! test#java#maventest#build_args(args) abort
-  let args = ['-Dtest='] + a:args
-  return [join(args, "")]
+  " In order to support custom options passed to mvn we need split
+  " the options from the test name (if one is passed).
+  let test_name = ''
+  let opts = []
+  let user_options = get(g:, 'test#java#maventest#options', '')
```

Comment:
You shouldn't need to read the `test#java#maventest#options` variable, its value is already merged into `a:args`.